### PR TITLE
WIP: Feature/cli beautify

### PR DESCRIPTION
**Pull request by sshaman1101**
Rewrite CLI using cobra

TODO:
- Fix all broken gRPC responses
- ~~Pretty print all outputs~~
- Write help messages with examples
- Read config from `$HOME`
- Read `$HOME/.docker/config.json` for registry credentials
- Refactoring?

### CODE REVIEW

**Comment by sshaman1101** [comment]
Fixed container delete; Now trying to fix task status;
Please review [insonmnia/miner/overseer.go:153](https://github.com/sonm-io/core/blob/feature/cli-beautify/insonmnia/miner/overseer.go#L153), this code is called on regular container deletion but crashes:
```
2017-08-04T16:46:42.979+0300	ERROR	miner/container.go:102	failed to remove the container
{"error": "Error response from daemon: removal of container 8f57a12abede6579d4fc6a008eee4f02910021166ff2c217370a07974fb86476 is already in progress"}
```

**Comment by sshaman1101** [comment]
The last thing I want to fix is task status fetching.
@3Hren @noxiouz @antmat please review my commits.

**Comment by noxiouz** [comment]
gonna look today

File: insonmnia/hub/server.go [hunk 1]
```diff
@@ -144,77 +140,74 @@ func (h *Hub) StartTask(ctx context.Context, request *pb.StartTaskRequest) (*pb.
...
 func (h *Hub) MinerStatus(ctx context.Context, request *pb.MinerStatusRequest) (*pbminer.TasksStatusReply, error) {
 	log.G(ctx).Info("handling MinerStatus request", zap.Any("req", request))
+
 	miner := request.Miner
-	h.mu.Lock()
-	mincli, ok := h.miners[miner]
-	h.mu.Unlock()
+	mincli, ok := h.getMinerByID(miner)
 	if !ok {
+		log.G(ctx).Error("Miner not fount", zap.String("miner", miner))
```
Comment (3Hren) [inline insonmnia/hub/server.go]: Typo. Also we log with lowercase.

File: insonmnia/hub/server.go [hunk 2]
```diff
@@ -144,77 +140,74 @@ func (h *Hub) StartTask(ctx context.Context, request *pb.StartTaskRequest) (*pb.
...
 		return nil, status.Errorf(codes.NotFound, "no such miner %s", miner)
 	}
 
+	var reply pbminer.TasksStatusReply
+	err := mincli.fetchStatuses()
+
+	if err != nil {
+		log.G(ctx).Error("Cannot get status", zap.Error(err))
```
Comment (3Hren) [inline insonmnia/hub/server.go]: Log.

File: insonmnia/hub/server.go [hunk 3]
```diff
@@ -144,77 +140,74 @@ func (h *Hub) StartTask(ctx context.Context, request *pb.StartTaskRequest) (*pb.
...
+		return nil, status.Errorf(codes.DataLoss, "Cannot get status")
+	}
+
 	mincli.status_mu.Lock()
-	reply := pbminer.TasksStatusReply{mincli.status_map}
+	reply = pbminer.TasksStatusReply{mincli.status_map}
```
Comment (3Hren) [inline insonmnia/hub/server.go]: Named struct construction? Otherwise vet can go crazy.

File: insonmnia/miner/container.go [hunk 4]
```diff
@@ -77,23 +77,22 @@ func (c *dcontainer) startContainer() error {
...
 	// TODO: add atomic flag to prevent duplicated remove
 	defer func() {
-		c.remove()
 		// release HTTP connections
 		c.cancel()
 	}()
-	log.G(c.ctx).Info("kill the container")
 
-	if err = c.client.ContainerKill(c.ctx, c.ID, "SIGKILL"); err != nil {
+	log.G(c.ctx).Info("kill the container")
```
Comment (3Hren) [inline insonmnia/miner/container.go]: Attach id.

File: insonmnia/miner/overseer.go [hunk 5]
```diff
@@ -308,17 +301,20 @@ func (o *overseer) Spawn(ctx context.Context, d Description) (status chan pb.Tas
 func (o *overseer) Stop(ctx context.Context, containerid string) error {
 	o.mu.Lock()
 
-	pr, cok := o.containers[containerid]
-	s, sok := o.statuses[containerid]
+	descriptor, dok := o.containers[containerid]
+	status, sok := o.statuses[containerid]
+	// todo: maybe we check sok && sok before calling delete() ?
```
Comment (3Hren) [inline insonmnia/miner/overseer.go]: No need to check.

File: insonmnia/miner/overseer.go [hunk 6]
```diff
@@ -308,17 +301,20 @@ func (o *overseer) Spawn(ctx context.Context, d Description) (status chan pb.Tas
...
+
 	if sok {
-		s <- pb.TaskStatus_FINISHED
+		status <- pb.TaskStatus_FINISHED
 	}
 
-	if !cok {
-		return fmt.Errorf("no such container %s", containerid)
+	if !dok {
+		return fmt.Errorf("no such container %status", containerid)
```
Comment (3Hren) [inline insonmnia/miner/overseer.go]: ?

Comment (sshaman1101) [inline insonmnia/miner/overseer.go]: OH SHI~, JetBrains smart refactoring no so smart :)

**Comment by 3Hren** [comment]
I'd like to rebase this, @noxiouz PTAL.

**Pull request merged**